app/app_aes.c

```diff
@@ -23,6 +27,642 @@ void app_aes_cleanup(void) {
+            goto err;
+        }
+        OSSL_PARAM_BLD_push_uint(pbld, OSSL_CIPHER_PARAM_PADDING, 0);
+        if (tc->cipher == ACVP_AES_CFB1) {
+            OSSL_PARAM_BLD_push_uint(pbld, OSSL_CIPHER_PARAM_USE_BITS, 1);
+        }
+        if (tc->cipher == ACVP_AES_CBC_CS2) {
+            OSSL_PARAM_BLD_push_utf8_string(pbld, OSSL_CIPHER_PARAM_CTS_MODE, "CS2", 0);
+        }
+        if (tc->cipher == ACVP_AES_CBC_CS3) {
+            OSSL_PARAM_BLD_push_utf8_string(pbld, OSSL_CIPHER_PARAM_CTS_MODE, "CS3", 0);
+        }
+        params = OSSL_PARAM_BLD_to_param(pbld);
+        if (!params) {
+            printf("Error generating params in AES\n");
+            goto err;
+        }
+        if (EVP_CipherInit_ex2(cipher_ctx, cipher, tc->key, iv, direction, params) != 1) {
+            printf("Error initializing cipher in AES\n");
+            goto err;
+        }
+
+        if (tc->direction == ACVP_SYM_CIPH_DIR_ENCRYPT) {
+            if (EVP_CipherUpdate(cipher_ctx, tc->ct, (int *)&tc->ct_len, tc->pt, tc->pt_len) != 1) {
+                printf("Error performing encrypt in AES\n");
+                goto err;
+            }
+            if (EVP_CipherFinal_ex(cipher_ctx, tc->ct, (int *)&tc->ct_len) != 1) {
+                printf("Error finalizing encrypt in AES\n");
+                goto err;
+            }
+            tc->ct_len = tc->pt_len;
+        } else if (tc->direction == ACVP_SYM_CIPH_DIR_DECRYPT) {
+            if (EVP_CipherUpdate(cipher_ctx, tc->pt, (int *)&tc->pt_len, tc->ct, tc->ct_len) != 1) {
+                printf("Error performing decrypt in AES\n");
+                goto err;
+            }
+            if (EVP_CipherFinal_ex(cipher_ctx, tc->pt, (int *)&tc->pt_len) != 1) {
+                printf("Error finalizing decrypt in AES\n");
+                goto err;
+            }
+            tc->pt_len = tc->ct_len;
+        } else {
+            printf("Unsupported direction\n");
+            goto err;
+        }
+
+        EVP_CIPHER_CTX_free(cipher_ctx);
+        glb_cipher_ctx = NULL;
+    }
+    rv = 0;
+
+err:
+    if (rv != 0) {
+        if (glb_cipher_ctx) EVP_CIPHER_CTX_free(glb_cipher_ctx);
+        glb_cipher_ctx = NULL;
+    }
+    if (cipher) EVP_CIPHER_free(cipher);
+    if (pbld) OSSL_PARAM_BLD_free(pbld);
+    if (params) OSSL_PARAM_free(params);
+    return rv;
+}
+
+int app_aes_keywrap_handler(ACVP_TEST_CASE *test_case) {
+    ACVP_SYM_CIPHER_TC *tc;
+    EVP_CIPHER_CTX *cipher_ctx = NULL;
+    EVP_CIPHER *cipher = NULL;
+    int out_len;
+    int rv = 1;
+    ACVP_SUB_AES alg;
+    char alg_name[32];
+    const char *type = NULL, *mode = NULL, *inv = NULL;
+
+    if (!test_case) {
+        return rv;
+    }
+
+    tc = test_case->tc.symmetric;
+
+    /* Begin encrypt code section */
+    cipher_ctx = EVP_CIPHER_CTX_new();
+    EVP_CIPHER_CTX_init(cipher_ctx);
+
+    alg = acvp_get_aes_alg(tc->cipher);
+    if (alg == 0) {
+        printf("Invalid cipher value");
+        return 1;
```

Comment:
goto err ?

Reply:
These were all some of the bits I just copied over. Fixed in new and old, thanks!